tiledb/proc-macro/src/option_subset.rs

```diff
@@ -0,0 +1,118 @@
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, Span};
+
+pub fn expand(input: &syn::DeriveInput) -> TokenStream {
+    let name = &input.ident;
+    let (impl_generics, ty_generics, where_clause) =
+        input.generics.split_for_impl();
+    let body = option_subset_body(input);
+
+    let expanded = quote! {
+        impl #impl_generics OptionSubset for #name #ty_generics #where_clause {
+            fn option_subset(&self, other: &Self) -> bool {
+                #body
+            }
+        }
+    };
+    expanded.into()
+}
+
+fn option_subset_fields(fields: &syn::Fields) -> proc_macro2::TokenStream {
```

Comment:
Super minor nit, but I prefer to sort functions by their call order. Currently we've got `expand -> option_subset_body -> option_subset_fields` as the call order, but the source code order is `expand -> open_subset_fields -> option_subset_body`.